ckanext/datagovuk/plugin.py

```diff
@@ -286,16 +286,14 @@ def before_create_or_update(self, context, resource):
         # If resource is an API, don't do anything special
         if resource.get("format") == "API":
             return
-        elif "upload" not in resource:
-            return
-        elif "upload" in resource and resource["upload"] == "":
+        elif resource.get("upload") is None:
```

Comment:
Might be clearer like this - 

`elif 'upload' not in resource:`

Reply:
See https://github.com/alphagov/ckanext-datagovuk/pull/237#discussion_r366789215